ssl/t1_lib.c

```diff
@@ -2451,7 +2436,7 @@ static int ssl_scan_serverhello_tlsext(SSL *s, PACKET *pkt, int *al)
         else if (type == TLSEXT_TYPE_application_layer_protocol_negotiation) {
             unsigned len;
             /* We must have requested it. */
-            if (s->alpn_client_proto_list == NULL) {
+            if (!s->s3->alpn_sent || s->alpn_client_proto_list == NULL) {
```

Comment:
During a renegotiation, the client does not send ALPN, even though it may have alpn_client_proto_list. So we want to make sure it was actually sent, before we accept it.
Also possible, that alpn_client_proto_list was updated in the middle of a handshake (less likely).